src/AdminBarApi.php

```diff
@@ -24,19 +80,55 @@ class AdminBarApi extends AbstractEndpoint {
 	 * @return array|\WP_Error
 	 */
 	public function endpoint_callback( \WP_REST_Request $request ) {
-		global $wp_admin_bar;
+		$admin_url = admin_url();
+		$edit_url = '';
+		$admin_bar_user = '';
+		$user_id = 0;
+
+		if ( isset( $_COOKIE[ $this->cookie_name ] ) ) { // Input var okay.
+			$admin_bar_user = sanitize_text_field( wp_unslash( $_COOKIE[ $this->cookie_name ] ) ); // Input var okay.
+		}
+
+		if ( '' !== $admin_bar_user ) {
+			$user = get_userdatabylogin( $admin_bar_user );
 
-		$current_user = wp_get_current_user();
+			if ( $user ) {
+				 $user_id = $user->ID;
+			}
+		}
+
+		if ( 0 !== $user_id && user_can( $user_id, 'edit_posts' ) ) {
```

Comment:
Here you need to check that the user can edit the current page's post_type. Because if they can edit posts, doesn't mean they can edit pages for example.
Mm but I guess you're adding the post_id on the frontend, which means you don't need to make a trip to the server each route change.
I think the best way would be to return a list of the tyes they can edit, so the fe can decide whether to show this menu or not.
However, in the interest of time let's do that later, I'll open an issue.